libraries/botbuilder-dialogs-adaptive-testing/src/testRunner.ts

```diff
@@ -16,34 +16,33 @@ import { TestScript } from './testScript';
  * In charge of running Dialog Adaptive tests.
  */
 export class TestRunner {
-    private resourceExplorer: ResourceExplorer;
-    private testAdapter: TestAdapter;
-
     /**
      * Initializes a new instance of the `TestRunner` class.
      * @param resourcePath Path where the `.dialog` files are located.
      */
-    public constructor(resourcePath: string) {
+    public constructor(private resourcePath: string) {
         ComponentRegistration.add(new AdaptiveComponentRegistration());
         ComponentRegistration.add(new AdaptiveTestComponentRegistration());
         ComponentRegistration.add(new QnAMakerComponentRegistration());
         ComponentRegistration.add(new LuisComponentRegistration());
-
-        this.resourceExplorer = new ResourceExplorer();
-        this.resourceExplorer.addFolder(resourcePath, true, false);
-
-        this.testAdapter = new TestAdapter(TestAdapter.createConversation('botbuilder-dialogs-adaptive-testing'));
     }
 
     /**
      * Runs a test script with the specified name.
      * @param testName Test name.
      * @returns A Promise that represents the work queued to execute.
      */
-    public async runTestScript(testName: string): Promise<any> {
-        const script = this.resourceExplorer.loadType<TestScript>(`${testName}.test.dialog`);
+    public runTestScript(testName: string): Promise<void> {
+        // Construct a fresh test adapter
+        const testAdapter = new TestAdapter(TestAdapter.createConversation('botbuilder-dialogs-adaptive-testing'));
+
+        // Construct a fresh resourceExplorer
+        const resourceExplorer = new ResourceExplorer();
+        resourceExplorer.addFolder(this.resourcePath, true, false);
+
+        const script = resourceExplorer.loadType<TestScript>(`${testName}.test.dialog`);
         script.description = script.description || testName;
-        this.testAdapter.activeQueue.splice(0, this.testAdapter.activeQueue.length);
```

Comment:
This is not great either, and is definitely a point for creating a new TestAdapter every time `runTestScript()` is called.

With this removal in mind, I'm leaning towards passing in `TestAdapter` constructor parameters might be the better option. Either through the `TestRunner` constructor or through `runTestScript()`. If TestAdapter ctor params are passed in instead, devs are still able to configure the TestAdapter to their liking.

The "TestAdapterCtorOptions" object would probably need a `middleware: Middleware|Array<Middleware>` property.

Reply:
My intention with this change was just to fix the event emitter leak; changing or refactoring signatures is out of scope for this PR. We can file an issue to clean up the external interface of the `TestRunner` separately if it's worth prioritizing.

Reply:
I think this is probably the best approach overall, tactical fix and tracking issue for the final design of the shipped product.